aptos-move/block-executor/src/scheduler.rs

```diff
@@ -452,9 +513,10 @@ impl Scheduler {
 
         if let TransactionStatus::Executing(incarnation) = *status {
             *status = TransactionStatus::Suspended(incarnation, dep_condvar);
-        } else {
-            unreachable!();
         }
+        // else {
```

Comment:
Did you observe this failing?

I see two differences in what we are doing now due to waking future waiting threads as it pertains to the scheduler.
(1) we aren't changing the status from ReadyToExecute / Suspended.
(2) we aren't calling finish_execution in executor (as an optimization).

There are indeed some ugly cases, like the thread leaves it in Suspended state, then real dependency marks it ready_to_execute, then someone tries to re-execute and notifies the already notified condvar :D I think it must be fine in terms of status invariants, but maybe the cleanest is to add a new "Halted" status. Then we can consider it in invariants, e.g. here match status, ignore Halted, do the same in Executing, and still unreachable for everything else.

Halted should basically be ignored by everybody as far as creating new tasks goes, it literally means task was executing/suspended and the whole execution is finished & assigned thread went home.